.circleci/config.yml

```diff
@@ -0,0 +1,120 @@
+version: 2.1
+orbs:
+  aws-cli: circleci/aws-cli@1.3.1
+
+commands:
+  check-for-changes:
+    description: "Searches for any files matching the pattern in a diff result"
+    parameters:
+      in:
+        description: "Returns true if this pattern matches any folder/file in the diff"
+        type: string
+      since:
+        description: "The git commit that we are checking against"
+        type: string
+    steps:
+      - run:
+          command: |
+            git diff --name-only HEAD << parameters.since >> | grep << parameters.in >>
+
+  write-build-info:
+    description: "Write relevant info to the current build/deploy to a file for referencing"
+    parameters:
+      data:
+        description: "The contents of the build file"
+        type: string
+        default: |
+          BUILD_COMMIT=<< pipeline.git.revision >>
+      dest:
+        description: "The destination of the build file"
+        type: string
+    steps:
+      - run:
+          name: "Writing build info"
+          command: |
+            echo "<< parameters.data >>" > << parameters.dest >>/BUILD.txt
+
+  copy-to-s3:
```

Comment:
There's also [this orb](https://circleci.com/developer/orbs/orb/circleci/aws-s3). I'm pretty ignorant here, though. Is it better to use an orb like this or run a simple `aws s3` command and assume Circle has all the context it needs?

@dginther?

Reply:
I did see that orb when I was looking at available options, but I haven't tried it.  I'm not _against_ trying it out though. The reason I went with the `aws-cli` orb is because it allowed for less abstraction, since I could have Circle run the exact same command that I (or someone else) would have run on the command line.